Update hapi to @hapi/hapi to fix shot dependency node deprecation error

Carthanial opened the pull request:
### Description : 

- Fixes node deprecation warning ` DeprecationWarning: OutgoingMessage.prototype._headers is deprecated`. 
- This error is due to a devDependency of the `hapi` module called `shot`. 
- This new version of hapi `@hapi/hapi` has the updated `@hapi/shot` module which no longer uses the `_headers` property and replaces it with the updated node http method `getHeaders()`

#### Note: 
This fix will cause errors in files that require in the module `hapi` which is the old version of `@hapi/hapi` and further PRs will be needed to fix those issues. 

##### Reference Issue #64

zhyuanqi commented:
Referencing this CR https://github.com/Tengda-He/kibana/pull/76/files , you also updated nock and added hapi/shot in devdependencies. Let's remove these as well

Carthanial commented:
> Referencing this CR https://github.com/Tengda-He/kibana/pull/76/files , you also updated nock and added hapi/shot in devdependencies for bfs-7.7-upgrade. Let's remove these as well

I have removed them locally so I will perform a force push to remove those commits.
